File: lobwebapp-html/src/main/scripts/lwa/service/contract/UserService.ts
///<reference path="./../../reference.d.ts"/>

module d.service.contract {
    export interface UserService extends d.service.contract.base.EntityService<domain.User> {

        findByUsername(username: string,
            successCallback: (data: domain.User, status: number, headers: (headerName: string) => string, config: ng.IRequestConfig) => any,
            errorCallback: (data: any, status: number, headers: (headerName: string) => string, config: ng.IRequestConfig) => any
            ): void;

    }

    export interface UserRole {
        admin: string;
        manager: string;
        salesman: string;
        client: string;
    }
}

File: lobwebapp-core/src/main/java/com/espindola/lobwebapp/service/contract/base/UserService.java
package com.espindola.lobwebapp.service.contract.base;


import com.espindola.lobwebapp.domain.User;
import com.espindola.lobwebapp.service.contract.EntityService;

public interface UserService extends EntityService<User> {
	User findByUsername(String username);
	Boolean authenticate(User user);
}
